### ISSUE TITLE
Using shorthand method name in constructor with super

### ISSUE THREAD

**Issue by Tigge** [Microsoft/ChakraCore#5030]
Using a shorthand method name in an object in a constructor with a super does not work in `Microsoft Edge 41.16299.371.0`, `Microsoft EdgeHTML 16.16299`. It results in a `Invalid use of the 'super' keyword` error.

Does not work (with shorthand method name):

```js
class A extends Object {
  constructor() {
    const b = {c() {}};
    super();
  }
};
```

Works (without shorthand method name):

```js
class A extends Object {
  constructor() {
    const b = {c: function() {}};
    super();
  }
};
```

**Comment by Tigge** [comment]
The same issue seems to happen on ChakraCore master as well:

```
SyntaxError: Invalid use of the 'super' keyword
	at code (test.js:4:10)
```

**Comment by sethbrenith** [comment]
Thanks for reporting!

**Comment by dilijev** [comment]
Confirmed repro. Looks like a parser issue for object-method syntax.

```
## Source
class A extends Object {
  constructor() {
    const b = {c() {}};
    super();
  }
};
print('done')


#### jsvu-ch
SyntaxError: Invalid use of the 'super' keyword

#### jsvu-d8, jsvu-jsc, jsvu-sm
done
```

**Comment by rhuanjl** [comment]
@dilijev I've made a PR to fix this #5036
